lib/prepare-write.js

```diff
@@ -1,43 +1,27 @@
 'use strict';
 
-var assign = require('object-assign');
 var path = require('path');
 var mkdirp = require('mkdirp');
 var fs = require('graceful-fs');
 
-function booleanOrFunc(v, file) {
-  if (typeof v !== 'boolean' && typeof v !== 'function') {
-    return null;
-  }
-
-  return typeof v === 'boolean' ? v : v(file);
-}
-
-function stringOrFunc(v, file) {
-  if (typeof v !== 'string' && typeof v !== 'function') {
-    return null;
-  }
-
-  return typeof v === 'string' ? v : v(file);
-}
+var normalizeOptions = require('./normalize-options');
 
 function prepareWrite(outFolder, file, opt, cb) {
-  var options = assign({
-    cwd: process.cwd(),
-    mode: (file.stat ? file.stat.mode : null),
-    dirMode: null,
-    overwrite: true,
-  }, opt);
-  var overwrite = booleanOrFunc(options.overwrite, file);
-  options.flag = (overwrite ? 'w' : 'wx');
+  var options = normalizeOptions(opt, {
+    cwd: { type: 'string', defaultValue: process.cwd() },
```

Comment:
Could you explain why you find it awkward?

As for the alternative solution, do you mean I should duplicate this line for every option? Like:

```
var options = {
  cwd: defaultValue(valueOrFunction.string(opt.cwd, file), process.cwd()),
  mode: defaultValue(valueOrFunction.number(opt.mode, file), (file.stat ? file.stat.mode : null)),
  dirMode: ...
};
```